src/Workspaces/Core/Portable/CodeGeneration/CodeGenerationHelpers.cs

```diff
@@ -102,6 +103,62 @@ public static bool IsSpecialType(ITypeSymbol type, SpecialType specialType)
             return type != null && type.SpecialType == specialType;
         }
 
+        /// <summary>
+        /// Indicates whether the specified attribute is <see cref="AllowNullAttribute"/>, <see cref="MaybeNullAttribute"/> or <see cref="MaybeNullWhenAttribute"/>.
+        /// </summary>
+        public static bool IsAllowNullAttribute(AttributeData attribute) =>
+            IsFlowAnalysisNamespace(attribute.AttributeClass.ContainingNamespace) &&
+                attribute.AttributeClass.Name switch
+                {
+                    nameof(AllowNullAttribute) => true,
+                    nameof(MaybeNullAttribute) => true,
+                    nameof(MaybeNullWhenAttribute) => true,
+                    _ => false
+                };
+
+        public static bool IsNullableFlowAnalysisAttribute(AttributeData attribute) =>
+            IsFlowAnalysisNamespace(attribute.AttributeClass.ContainingNamespace) &&
+                attribute.AttributeClass.Name switch
+                {
+                    nameof(AllowNullAttribute) => true,
+                    nameof(DisallowNullAttribute) => true,
+                    nameof(MaybeNullAttribute) => true,
+                    nameof(NotNullAttribute) => true,
+                    nameof(MaybeNullWhenAttribute) => true,
+                    nameof(NotNullWhenAttribute) => true,
+                    nameof(NotNullIfNotNullAttribute) => true,
+                    _ => false
+                };
+
+        private static bool IsFlowAnalysisNamespace(INamespaceSymbol namespaceSymbol)
+        {
+            return GetNameParts(namespaceSymbol).SequenceEqual(
+                new[]
+                {
+                    nameof(System),
+                    nameof(System.Diagnostics),
+                    nameof(System.Diagnostics.CodeAnalysis)
+                });
+
+            static List<string> GetNameParts(INamespaceSymbol namespaceSymbol)
```

Comment:
1. i think we have this helper somewhere already.
2. we should really support this without having to allocate.

Reply:
Fixed per @jasonmalinowski's suggestion.